pkgs/development/python-modules/xgboost/default.nix

```diff
@@ -10,28 +10,52 @@
 , matplotlib
 , graphviz
 , datatable
+, hypothesis
 }:
 
 buildPythonPackage {
   pname = "xgboost";
   inherit (xgboost) version src meta;
 
-  patches = [
-    (substituteAll {
-      src = ./lib-path-for-python.patch;
-      libpath = "${xgboost}/lib";
-      extention = stdenv.hostPlatform.extensions.sharedLibrary;
-    })
+  nativeBuildInputs = [ cmake ];
+  buildInputs = [ xgboost ];
+  propagatedBuildInputs = [ scipy ];
+  checkInputs = [
+    pytestCheckHook
+    scikitlearn
+    pandas
+    matplotlib
+    graphviz
+    datatable
+    hypothesis
   ];
 
-  postPatch = "cd python-package";
+  # Override existing logic for locating libxgboost.so which is not appropriate for Nix
+  prePatch = let
+    libPath = "${xgboost}/lib/libxgboost${stdenv.hostPlatform.extensions.sharedLibrary}";
+  in ''
+    echo 'find_lib_path = lambda: ["${libPath}"]' > python-package/xgboost/libpath.py
+  '';
```

Comment:
hmmm cc @FRidh 

How do you feel about this type of patching

I could see issues in the future if the filename changes... but those would also be issues with the current patching paradigm as well

Reply:
The prior patch file stopped working when I updated to 1.4.1, and it basically just removed this entire file anyways: https://github.com/dmlc/xgboost/blob/master/python-package/xgboost/libpath.py#L14-L65. I figured this was easier for future maintainers than creating a new patch file.

Reply:
Yea, I can see the merits and appeal of doing it this way. But fridh has much more historical context, and would like his opinion; if he's okay with it, I think we can move forward.